### DISCUSSION TITLE
shapefile filtering

### DISCUSSION THREAD

**Discussion by dieterdreist** [systemed/tilemaker#404] [Q&A] [answered]
I am using tilemaker with a set of shapefiles, and it is working quite well. But there are some features in the shapes that I would not want in the output, and more importantly, that I would only want to have in higher zoom levels.
I am successfully using the attribute_function(attr,layer) to normalize attributes, but I could not manage to skip features. Is it possible?

**Comment by systemed** [comment, marked as answer]
Not at present I'm afraid, but I agree it's a good idea and it might be reasonably easy to implement - I'll give it some thought.

Ultimately I would like to bring the shapefile Lua interface closer to the main (OSM-derived) PBF interface so that you can write objects out to whatever layer you want, as centroids if necessary, etc.
